docs/design/adminconsole/WORKER-CRON-PROCEES.md

```diff
@@ -0,0 +1,202 @@
+FROM mcr.microsoft.com/dotnet/sdk:8.0.401-alpine3.20@sha256:658c93223111638f9bb54746679e554b2cf0453d8fb7b9fed32c3c0726c210fe AS build
+  
+WORKDIR /source
+
+# Copy source code and compile the application
+COPY Application/EdFi.AdminConsole.HealthCheckService/. ./EdFi.AdminConsole.HealthCheckService/
+
+WORKDIR /source/EdFi.AdminConsole.HealthCheckService
+
+# Restore dependencies, Then build and publish release
+RUN dotnet restore &&\
+  dotnet publish -c Release -o /app
+
+# .NET Runtime image to execute the application
+FROM mcr.microsoft.com/dotnet/runtime:8.0.11 AS runtime
+
+# Install Cron.
+RUN apt-get update && apt-get install -y cron
+
+# Add cron from file and adjust permissions
+COPY crontab /etc/cron.d/container_cronjob
+RUN chmod 0644 /etc/cron.d/container_cronjob
+RUN crontab /etc/cron.d/container_cronjob  
+WORKDIR /app
+
+# Add Published executable
+COPY --from=build /app .
+
+# Execute the app via chron
+CMD ["cron", "-f"]
+```
+
+### Add Docker compose for Development
+
+The `health-check-svs.yml` docker compose file provides a simple way to test building the image and logging the output.
+
+This might be moved or modified in the future to better support a multi-container environment.
+
+```yml
+version: "3.9"
+
+services:
+  health-check-service:
+    build:
+      context: ../
+      dockerfile: Docker/Dockerfile
+    volumes:
+      - service-logs:/var/log/achealthsvc.log
+    env_file:
+      - .env
+
+volumes:
+  service-logs:
+    name: health-check-service-logs
+
+```
+
+## Testing Strategy
+
+The Docker file takes advantage of crontab file by piping the output of th cron executable to a log file. To ensure the executable is running, this log file is inspected to ensure the output from running the application is piped to the `achealthsvc.log` log file.
+
+To verify the schedule, a user can build the image using the Dockerfile and docker-compose included in the `/Docker/` directory.
+
+`docker compose -f .\Docker\health-check-svc.yml up`
+
+Can check that the job is in the table of jobs to run
+`$ crontab -l`
+
+To verify the environment variables can be seen, in the container, once can use the `env` command from a connected shell, and inspect the values returned.
+
+## Additional Notes
+
+This holds any additional questions or comments that do not fit into the design spec above.
+
+### Outstanding Questions
+
+Will The Instance management worker need to be executed on a cron schedule? What would be executed?
+
+What is the difference between Admin API Client and Admin API Caller?
+
+How will the container image call the other services when hosted on localhost in other execution contexts?
+
+What is the expected deployment / execution environment for these services?
+
+* Docker compose okay when developing
+* How will the workers be incorporated into the deployment
+
+Is Multi-tenancy something that needs to be accounted for?
+
+Can the crontab file read an environment variable?
```

Comment:
Log management is also outstanding. The log fill will disappear as soon as the container stops running. If the container is running for a very long time, it could eventually become a very large log file. Serilog's built in log rotation might be all you need for handling this.

Reply:
Oh, you did create a volume mount in the sample docker compose. Great!